### PR TITLE
Create an API for enabling and disabling installed signal types

### PR DESCRIPTION
**Pull request by evank28**
- feat: add model for signal type config
- Create API for creating and updating signal type overrides

Summary
---------

* Database fields to store override config of signal types
* API endpoint to update enabled_ratio of any installed signal type, and to read all signal types
 

Test Plan
---------

```
vscode ➜ /workspace $ curl 'localhost:5000/c/signal_type'
[
  {
    "enabled_ratio": 0.5,
    "name": "pdq"
  },
  {
    "enabled_ratio": 1.0,
    "name": "video_md5"
  }
]
vscode ➜ /workspace $ curl -X PUT -H "Content-Type: application/json" -d '{"enabled_ratio": 0.8}' 
'localhost:5000/c/signal_type/pdq'
vscode ➜ /workspace $ curl 'localhost:5000/c/signal_type'
[
  {
    "enabled_ratio": 0.8,
    "name": "pdq"
  },
  {
    "enabled_ratio": 1.0,
    "name": "video_md5"
  }
]
vscode ➜ /workspace $ curl -X PUT -H "Content-Type: application/json" -d '{"enabled_ratio": 0.9}' 
'localhost:5000/c/signal_type/pdq'
vscode ➜ /workspace $ curl 'localhost:5000/c/signal_type'
[
  {
    "enabled_ratio": 0.9,
    "name": "pdq"
  },
  {
    "enabled_ratio": 1.0,
    "name": "video_md5"
  }
]
```

### CODE REVIEW

**Review by Dcallies — CHANGES_REQUESTED** [review]

File: open-media-match/src/OpenMediaMatch/storage/interface.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class SignalTypeConfig:
 
     # Signal types that are not enabled should not be used in hashing/matching
     enabled: bool
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/storage/interface.py:63]: blocking; Remove this guy - if we want to retain a way to resolve the coinflip, we could add it as a method

File: open-media-match/src/OpenMediaMatch/storage/interface.py [hunk 2]
```diff
@@ -71,6 +72,26 @@ class ISignalTypeConfigStore(metaclass=abc.ABCMeta):
...
+
+    @t.final
+    def create_or_update_signal_type_override(
+        self, signal_type: str, enabled_ratio: float
+    ) -> None:
+        """Update enabled ratio of an installed signal type."""
+        installed_signal_types = self.get_signal_type_configs()
+        if signal_type not in installed_signal_types:
+            raise ValueError(f"Unknown signal type {signal_type}")
+        if not (0.0 <= enabled_ratio <= 1.0):
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/storage/interface.py:89]: blocking: Move this to the database verifier (see the bank name validator for how to do this)

**Review by Dcallies — CHANGES_REQUESTED** [review]

File: open-media-match/src/OpenMediaMatch/database.py [hunk 3]
```diff
@@ -187,3 +187,14 @@ def deserialize_index(self) -> SignalTypeIndex[int]:
...
+
+
+class SignalTypeOverride(db.Model):  # type: ignore[name-defined]
+    """
+    Stores signal types and whether they are enabled or disabled.
+    By default, any type not in this database is
+    """
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(String(255), unique=True, index=True)
```
[the remark this answers is not in the document]
Comment (evank28) [reply open-media-match/src/OpenMediaMatch/database.py:199]: Thought about it, but it sounded like i might run into issues so I decided to just not be fancy - https://stackoverflow.com/questions/517579/strings-as-primary-keys-in-mysql-database

File: open-media-match/src/OpenMediaMatch/blueprints/curation.py [hunk 4]
```diff
@@ -304,10 +304,14 @@ def update_signal_type_config(signal_type_name: str):
...
-        "enabled": true,
+        "enabled_ratio": 1.0,
     }
     """
-    abort(501, "unimplemented")
+    enabled_ratio = utils.str_to_type(utils.require_json_param("enabled_ratio"), float)
+    persistence.get_storage().create_or_update_signal_type_override(
+        signal_type_name, enabled_ratio
+    )
+    return jsonify({"name": signal_type_name, "enabled_ratio": enabled_ratio}), 204
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/blueprints/curation.py:314]: jsonify not required here

**Review by Dcallies — APPROVED** [review]
shipitquick

**Pull request merged**